src/waveform/renderers/waveformwidgetrenderer.cpp

```diff
@@ -123,18 +124,20 @@ void WaveformWidgetRenderer::onPreRender(VSyncThread* vsyncThread) {
     double visualSamplePerPixel = m_zoomFactor * m_rateRatio / m_scaleFactor;
     m_visualSamplePerPixel = math_max(0.01, visualSamplePerPixel);
 
-    TrackPointer pTrack(m_pTrack);
-    ConstWaveformPointer pWaveform = pTrack ? pTrack->getWaveform() : ConstWaveformPointer();
-    if (pWaveform) {
-        m_audioSamplePerPixel = m_visualSamplePerPixel * pWaveform->getAudioVisualRatio();
-    } else {
-        m_audioSamplePerPixel = 0.0;
+    TrackPointer pTrack = m_pTrack;
+    if (pTrack) {
+        ConstWaveformPointer pWaveform = pTrack->getWaveform();
+        if (pWaveform) {
+            m_audioVisualRatio = pWaveform->getAudioVisualRatio();
+        }
     }
 
+    m_audioSamplePerPixel = m_visualSamplePerPixel * m_audioVisualRatio;
+
     double truePlayPos = m_visualPlayPosition->getAtNextVSync(vsyncThread);
-    // m_playPos = -1 happens, when a new track is in buffer but m_visualPlayPosition was not updated
+    // ruePlayPos = -1 happens, when a new track is in buffer but m_visualPlayPosition was not updated
 
```

Comment:
typo truePlayPos

Reply:
I have fixed it by a squash commit. I hope it is OK in this case.

Reply:
sure :)